src/main/java/github/scarsz/discordsrv/hooks/chat/HerochatProHook.java

```diff
@@ -0,0 +1,102 @@
+import com.herochat.api.Chatter;
+import com.herochat.HeroChat;
+
+/* DiscordSRV Imports */
+import github.scarsz.discordsrv.DiscordSRV;
+import github.scarsz.discordsrv.util.LangUtil;
+import github.scarsz.discordsrv.util.MessageUtil;
+import github.scarsz.discordsrv.util.PlayerUtil;
+import github.scarsz.discordsrv.util.PluginUtil;
+
+/* Other Imports */
+import net.kyori.adventure.text.Component;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.plugin.Plugin;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class HerochatProHook implements ChatHook {
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onMessage(ChannelChatEvent event) {
+        // make sure message isn't just blank
+        if (StringUtils.isBlank(event.getMessage())) return;
+
+        DiscordSRV.getPlugin().processChatMessage(event.getSender().getPlayer(), event.getMessage(), event.getChannel().getName(), event.getResult() != Chatter.Result.ALLOWED);
+    }
+
+    @Override
+    public void broadcastMessageToChannel(String channel, Component message) {
+        Channel chatChannel = getChannelByCaseInsensitiveName(channel);
+        if (chatChannel == null) return; // no suitable channel found
+        String legacy = MessageUtil.toLegacy(message);
+
+        String plainMessage = LangUtil.Message.CHAT_CHANNEL_MESSAGE.toString()
+                .replace("%channelname%", chatChannel.getName())
+                .replace("%channelnickname%", chatChannel.getNick())
+                .replace("%message%", legacy)
+                .replace("%channelcolor%", chatChannel.getColor().toString());
+
+        String translatedMessage = MessageUtil.translateLegacy(plainMessage);
+        chatChannel.sendRawMessage(translatedMessage);
+
+        PlayerUtil.notifyPlayersOfMentions(player ->
+                        chatChannel.getMembers().stream()
+                                .map(Chatter::getPlayer)
+                                .collect(Collectors.toList())
+                                .contains(player),
+                legacy);
+    }
+
+    private static Channel getChannelByCaseInsensitiveName(String name) {
+        List<Channel> channels = Herochat.getChannelManager().getChannels();
+
+        if (channels.size() > 0) {
+            for (Channel channel : Herochat.getChannelManager().getChannels()) {
+                DiscordSRV.debug("\"" + channel.getName() + "\" equalsIgnoreCase \"" + name + "\" == " + channel.getName().equalsIgnoreCase(name));
+                if (channel.getName().equalsIgnoreCase(name)) {
+                    return channel;
+                }
+            }
+            DiscordSRV.debug("No matching Herochat channels for name \"" + name + "\"");
+        } else {
+            DiscordSRV.debug("Herochat's channel manager returned no registered channels");
+        }
+        return null;
+    }
+
```

Comment:
isEnabled should be overridden, and it should check if the HerochatPro classes are available - if they are returning true otherwise false.
The same should be done for the regular Herochat hook, except checking if non-pro classes are available

Reply:
Vankka wasn't saying that the classes should be merged, just that the ChatHook#isEnabled method should be overridden to be more context sensitive for HeroChatPro

Reply:
Most of the Code above is just written out using the HerochatHook.java from DiscordSRV
If there’s any better implementation, or changes that you’d like to make to the above, feel free to do so. I was only updating the imports for the v6 plugin in the file above (: @Scarsz